### PR TITLE
Add different ui rendering methods blog

### PR DESCRIPTION
**Pull request by nurRiyad**

### CODE REVIEW

File: content/post/different-ui-rendering-methods/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  - bytebuilders
+  - javascript
+  - typescript
+  - vue
+  - nuxt
+  - byte.builders
+  - console.byte.builders
+  - kubedb.byte.builders
+  - learn.byte.builders
+  - selfhost.byte.builders
```
Comment (s4k1b) [inline content/post/different-ui-rendering-methods/index.md:19]: add csr, ssr, ssg, ur, spa

Comment (s4k1b) [inline content/post/different-ui-rendering-methods/index.md:19]: not needed

File: content/post/different-ui-rendering-methods/index.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+#### When to use SSG:
+
+- Building content-heavy websites, blogs, or documentation sites where content updates are infrequent.
+- Requiring exceptional performance and scalability while reducing server-side complexity.
+- Prioritizing security by minimizing the attack surface.
+
+### 4. Universal Rendering (CSR + SSR):
+
+In universal rendering when the browser requests a URL with universal (server-side + client-side) rendering enabled, the server returns a fully rendered HTML page to the browser. Whether the page has been generated in advance and cached or is rendered on the fly, at some point, the browser runs the JavaScript code in a server environment, producing an HTML document. Users immediately get the content of our application, contrary to client-side rendering. This step is similar to traditional server-side rendering performed by PHP or Ruby applications.
```
Comment (s4k1b) [inline content/post/different-ui-rendering-methods/index.md:83]: "the browser runs the JavaScript code in a server environment"

what?

File: content/post/different-ui-rendering-methods/index.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+To not lose the benefits of the client-side rendering method, such as dynamic interfaces and page transitions, the Client (browser) loads the JavaScript code that runs on the Server in the background once the HTML document has been downloaded. The browser interprets it again (hence Universal rendering) and CSR takes control of the document and enables interactivity.
+
+Universal rendering allows a Nuxt application to provide quick page load times while preserving the benefits of client-side rendering. Furthermore, as the content is already present in the HTML document, crawlers can index it without the overhead.
```
Comment (s4k1b) [inline content/post/different-ui-rendering-methods/index.md:87]: don't need to mention Nuxt here

File: content/post/different-ui-rendering-methods/index.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Out Projects
+
+We have several web project to render. Some of them require high interactivity, some of then need SEO and others are content heavy. Depends on the project requirement to use different rendering method in our project. Here is a table that shows in which project we use which UI rending methods.
```
Comment (s4k1b) [inline content/post/different-ui-rendering-methods/index.md:91]: Change the first sentence to "Here at Appscode, we have developed several web applications that utilize the different rendering techniques mentioned here."

**Review by s4k1b — CHANGES_REQUESTED** [review]

**Review by DRoyyy — DISMISSED** [review]

**Review by s4k1b — APPROVED** [review]

**Review by s4k1b — APPROVED** [review]

**Review by DRoyyy — APPROVED** [review]

**Pull request merged**